torchtune/modules/attention.py

```diff
@@ -156,23 +160,18 @@ def forward(
 
         TODO:
             - Return the attention weights
-            - Make application of positional embeddings optional
         """
         # input has shape [b, s, d]
         bsz, seq_len, _ = x.shape
-
-        if seq_len > self.max_seq_len:
```

Comment:
This felt like the wrong place to have this check as the attention layer itself doesn't care about max_seq_len. I moved it to the TransformerDecoder since the top level model cares, but maybe it would make more sense to move it to the rope module itself?